packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -71,53 +77,53 @@ export interface IPendingDetachedContainerState extends SnapshotWithBlobs {
 	pendingRuntimeState?: unknown;
 }
 
-interface SnapshotInfo extends SnapshotWithBlobs {
+export interface ISnapshotInfo extends SnapshotWithBlobs {
 	snapshotSequenceNumber: number;
 }
 
+export type ISerializedStateManagerDocumentStorageService = Pick<
+	IDocumentStorageService,
+	"getSnapshot" | "getSnapshotTree" | "getVersions" | "readBlob"
+> & {
+	loadedGroupIdSnapshots: Record<string, ISnapshot> | undefined;
+};
+
 export class SerializedStateManager {
 	private readonly processedOps: ISequencedDocumentMessage[] = [];
-	private snapshot: SnapshotWithBlobs | undefined;
+	private snapshot: ISnapshotInfo | undefined;
 	private readonly mc: MonitoringContext;
-	private latestSnapshot: SnapshotInfo | undefined;
-	private refreshSnapshot: Promise<void> | undefined;
+	private latestSnapshot: ISnapshotInfo | undefined;
+	private _initialRefreshSnapshot: Promise<void> | undefined;
 
 	constructor(
 		private readonly pendingLocalState: IPendingContainerState | undefined,
 		subLogger: ITelemetryLoggerExt,
-		private readonly storageAdapter: Pick<
-			IDocumentStorageService,
-			"readBlob" | "getSnapshotTree" | "getSnapshot" | "getVersions"
-		>,
-		private readonly _offlineLoadEnabled: boolean,
-		private readonly newSnapshotFetched?: () => void,
+		private readonly storageAdapter: ISerializedStateManagerDocumentStorageService,
+		public readonly offlineLoadEnabled: boolean,
+		private readonly supportGetSnapshotApi: boolean,
```

Comment:
What happened to newSnasphotFetched ?

Reply:
I believe you're gonna need to change refreshSerializedStateManager.spec.ts tests as well in case you're now use initialRefreshSnapshot

Reply:
This way, no one just throws random code into `newSnapshotFetched`, there's no need if the only thing we are doing is waiting on the new snapshot getting fetched.